# Adding Coordinates to the Datafrane using geocode from Google

### Calling the necessary dependencies

In [1]:
# Dependencies
import pandas as pd
import numpy as np
import requests
import json
import pprint

# Google API Key
from config import gkey

### Import the census data by County

In [4]:
coordinates_pd = pd.read_csv("../1_census/output/census_data_county.csv")
coordinates_pd

State                                  Name  Population  Median Age  \
0        72        Corozal Municipio, Puerto Rico     34933.0        38.1   
1        72        Maunabo Municipio, Puerto Rico     11297.0        43.5   
2        72       Peñuelas Municipio, Puerto Rico     21661.0        36.4   
3        72          Ponce Municipio, Puerto Rico    148863.0        39.5   
4        72  San Sebastián Municipio, Puerto Rico     38970.0        42.0   
...     ...                                   ...         ...         ...   
3215     55                Iowa County, Wisconsin     23576.0        42.5   
3216     55               Pepin County, Wisconsin      7282.0        45.8   
3217     55             Ozaukee County, Wisconsin     87817.0        43.9   
3218     55             Portage County, Wisconsin     70371.0        36.6   
3219     55            Richland County, Wisconsin     17626.0        44.8   

      Household Income  Per Capita Income  Poverty Count  Poverty Rate  \
0              14752.0             7887.0        19146.0     54.807775   
1              17636.0             8830.0         5945.0     52.624591   
2              16868.0             7983.0        12379.0     57.148793   
3              16561.0            10775.0        75187.0     50.507514   
4              14275.0             8072.0        21472.0     55.098794   
...                ...                ...            ...           ...   
3215           60017.0            31717.0         2100.0      8.907363   
3216           51470.0            27901.0          938.0     12.881077   
3217           80526.0            45820.0         5087.0      5.792728   
3218           54620.0            28363.0         8757.0     12.444047   
3219           48234.0            24941.0         2349.0     13.326903   

      Unemployment Rate  Households with Internet  \
0              9.435205                   11026.0   
1              6.506152                    3976.0   
2              7.391164                    7132.0   
3              4.711043                   53341.0   
4              8.486015                   14278.0   
...                 ...                       ...   
3215           1.895996                    9799.0   
3216           1.757759                    3005.0   
3217           1.637496                   35044.0   
3218           2.569240                   28072.0   
3219           1.753092                    7564.0   

      Households with Just Cell Data Plan  Households with DSL or Fiber Optic  \
0                                  2381.0                              2268.0   
1                                   434.0                              1078.0   
2                                  1361.0                              2329.0   
3                                  5613.0                             18366.0   
4                                  1525.0                              2446.0   
...                                   ...                                 ...   
3215                               1005.0                              5301.0   
3216                                224.0                              1969.0   
3217                               1978.0                             26993.0   
3218                               2469.0                             19121.0   
3219                                531.0                              3527.0   

      Households with Broadband Any Type  Households with Celular Data Plan  \
0                                 5021.0                             4088.0   
1                                 1577.0                              987.0   
2                                 3877.0                             3090.0   
3                                25211.0                            17691.0   
4                                 4743.0                             2737.0   
...                                  ...                                ...   
3215                         

### Create new columns to store the coordinate in the dataframe

In [52]:
# Add columns for lat, lng
coordinates_pd["Lat"] = 0
coordinates_pd["Lng"] = 0
coordinates_pd.head()

State                                  Name  Population  Median Age  \
0     72        Corozal Municipio, Puerto Rico     34933.0        38.1   
1     72        Maunabo Municipio, Puerto Rico     11297.0        43.5   
2     72       Peñuelas Municipio, Puerto Rico     21661.0        36.4   
3     72          Ponce Municipio, Puerto Rico    148863.0        39.5   
4     72  San Sebastián Municipio, Puerto Rico     38970.0        42.0   

   Household Income  Per Capita Income  Poverty Count  Poverty Rate  \
0           14752.0             7887.0        19146.0     54.807775   
1           17636.0             8830.0         5945.0     52.624591   
2           16868.0             7983.0        12379.0     57.148793   
3           16561.0            10775.0        75187.0     50.507514   
4           14275.0             8072.0        21472.0     55.098794   

   Unemployment Rate  Households with Internet  \
0           9.435205                   11026.0   
1           6.506152                    3976.0   
2           7.391164                    7132.0   
3           4.711043                   53341.0   
4           8.486015                   14278.0   

   Households with Just Cell Data Plan  Households with DSL or Fiber Optic  \
0                               2381.0                              2268.0   
1                                434.0                              1078.0   
2                               1361.0                              2329.0   
3                               5613.0                             18366.0   
4                               1525.0                              2446.0   

   Households with Broadband Any Type  Households with Celular Data Plan  \
0                              5021.0                             4088.0   
1                              1577.0                              987.0   
2                              3877.0                             3090.0   
3                             25211.0                            17691.0   
4                              4743.0                             2737.0   

   Households with Just DSL or Fiber Optic  \
0                                    706.0   
1                                    512.0   
2                                    736.0   
3                                   6802.0   
4                                   1335.0   

   Households with Satellite Internet  \
0                               361.0   
1                                77.0   
2                               181.0   
3                              1210.0   
4                               829.0   

   Households with Just Satellite Internet  Households with No Internet  Lat  \
0                                    161.0                       5686.0    0   
1                                     37.0                       2157.0    0   
2                                     27.0                       2972.0    0   
3                                    510.0                      25042.0    0   
4                                    456.0                       8622.0    0   

   Lng  
0    0  
1    0  
2    0  
3    0  
4    0

### Loop to consult the coordinates using the maps API

#### Warning: Looping through 3,219 rows takes about 15 minutes

In [53]:
# create a params dict that will be updated with new city each iteration
params = {"key": gkey}

#i = 0

# Loop through the cities_pd and run a lat/long search for each city
for index, row in coordinates_pd.iterrows():
#    if i <= 1:
        base_url = "https://maps.googleapis.com/maps/api/geocode/json"

        county = row['Name']

        # update address key value
        params['address'] = f"{county}"

        # make request
        county_lat_lng = requests.get(base_url, params=params)
        
        # print(county_lat_lng.url)
    
        # convert to json
        county_lat_lng = county_lat_lng.json()

        # pprint.pprint(county_lat_lng)
        
        # print(county_lat_lng['results'][0]["geometry"]["location"]["lat"])
        try:
            coordinates_pd.loc[index, "Lat"] = county_lat_lng['results'][0]["geometry"]["location"]["lat"]
            coordinates_pd.loc[index, "Lng"] = county_lat_lng['results'][0]["geometry"]["location"]["lng"]
            print(f"Retrieving information for '{county}'")
        except (KeyError, IndexError):
            print("Missing field/result... skipping.")
        print("-------------------------------------------------------------------------")
#    i += 1
# Visualize to confirm lat lng appear


Retrieving information for 'Corozal Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Maunabo Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Peñuelas Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Ponce Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'San Sebastián Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Toa Alta Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Toa Baja Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Vieques Municipio, Puerto R

Retrieving information for 'Gurabo Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Salinas Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Barceloneta Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Loíza Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Yauco Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Guaynabo Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Aguadilla Municipio, Puerto Rico'
-------------------------------------------------------------------------
Retrieving information for 'Cabo Rojo Municipio, Puerto Rico

Retrieving information for 'Perry County, Alabama'
-------------------------------------------------------------------------
Retrieving information for 'Walker County, Alabama'
-------------------------------------------------------------------------
Retrieving information for 'Cleburne County, Alabama'
-------------------------------------------------------------------------
Retrieving information for 'Greene County, Alabama'
-------------------------------------------------------------------------
Retrieving information for 'Barbour County, Alabama'
-------------------------------------------------------------------------
Retrieving information for 'Bibb County, Alabama'
-------------------------------------------------------------------------
Retrieving information for 'Bullock County, Alabama'
-------------------------------------------------------------------------
Retrieving information for 'Calhoun County, Alabama'
----------------------------------------------------------------

Retrieving information for 'Pinal County, Arizona'
-------------------------------------------------------------------------
Retrieving information for 'White County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Nevada County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'St. Francis County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Pulaski County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Clay County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Cross County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Dallas County, Arkansas'
----------------------------------------------------------

Retrieving information for 'Lawrence County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Monroe County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Polk County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Conway County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Ouachita County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Newton County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Arkansas County, Arkansas'
-------------------------------------------------------------------------
Retrieving information for 'Little River County, Arkansas'
------------------------------------------------

Retrieving information for 'Sutter County, California'
-------------------------------------------------------------------------
Retrieving information for 'San Mateo County, California'
-------------------------------------------------------------------------
Retrieving information for 'Tuolumne County, California'
-------------------------------------------------------------------------
Retrieving information for 'San Joaquin County, California'
-------------------------------------------------------------------------
Retrieving information for 'Amador County, California'
-------------------------------------------------------------------------
Retrieving information for 'Costilla County, Colorado'
-------------------------------------------------------------------------
Retrieving information for 'Adams County, Colorado'
-------------------------------------------------------------------------
Retrieving information for 'Conejos County, Colorado'
------------------------------------

Retrieving information for 'Ouray County, Colorado'
-------------------------------------------------------------------------
Retrieving information for 'Rio Blanco County, Colorado'
-------------------------------------------------------------------------
Retrieving information for 'Sedgwick County, Colorado'
-------------------------------------------------------------------------
Retrieving information for 'Weld County, Colorado'
-------------------------------------------------------------------------
Retrieving information for 'New Haven County, Connecticut'
-------------------------------------------------------------------------
Retrieving information for 'Litchfield County, Connecticut'
-------------------------------------------------------------------------
Retrieving information for 'Hartford County, Connecticut'
-------------------------------------------------------------------------
Retrieving information for 'Windham County, Connecticut'
---------------------------------

Retrieving information for 'Hardee County, Florida'
-------------------------------------------------------------------------
Retrieving information for 'Collier County, Florida'
-------------------------------------------------------------------------
Retrieving information for 'Okeechobee County, Florida'
-------------------------------------------------------------------------
Retrieving information for 'Orange County, Florida'
-------------------------------------------------------------------------
Retrieving information for 'Alachua County, Florida'
-------------------------------------------------------------------------
Retrieving information for 'District of Columbia, District of Columbia'
-------------------------------------------------------------------------
Retrieving information for 'Pasco County, Florida'
-------------------------------------------------------------------------
Retrieving information for 'Gilchrist County, Florida'
--------------------------------------

Retrieving information for 'Madison County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Mitchell County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Seminole County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Tift County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Twiggs County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Polk County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Talbot County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Taylor County, Georgia'
-----------------------------------------------------------------

Retrieving information for 'Lumpkin County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'McDuffie County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Miller County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Morgan County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Heard County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Oconee County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Dougherty County, Georgia'
-------------------------------------------------------------------------
Retrieving information for 'Fannin County, Georgia'
-------------------------------------------------------------

Retrieving information for 'Twin Falls County, Idaho'
-------------------------------------------------------------------------
Retrieving information for 'Owyhee County, Idaho'
-------------------------------------------------------------------------
Retrieving information for 'Blaine County, Idaho'
-------------------------------------------------------------------------
Retrieving information for 'Boise County, Idaho'
-------------------------------------------------------------------------
Retrieving information for 'Boundary County, Idaho'
-------------------------------------------------------------------------
Retrieving information for 'Butte County, Idaho'
-------------------------------------------------------------------------
Retrieving information for 'Caribou County, Idaho'
-------------------------------------------------------------------------
Retrieving information for 'Gooding County, Idaho'
-------------------------------------------------------------------------
Re

Retrieving information for 'Cumberland County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Ford County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Jo Daviess County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Kane County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Kendall County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Lawrence County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Brown County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Marion County, Illinois'
----------------------------------------------------

Retrieving information for 'Tazewell County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Cass County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Warren County, Illinois'
-------------------------------------------------------------------------
Retrieving information for 'Monroe County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'Fountain County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'Elkhart County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'Starke County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'Crawford County, Indiana'
----------------------------------------------------------

Retrieving information for 'Scott County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'Harrison County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'Huntington County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'LaPorte County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'Wabash County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'White County, Indiana'
-------------------------------------------------------------------------
Retrieving information for 'Jasper County, Indiana'
-------------------------------------------------------------------------
Missing field/result... skipping.
-------------------------------------------------------------------------
Retri

Retrieving information for 'Carroll County, Iowa'
-------------------------------------------------------------------------
Retrieving information for 'Cedar County, Iowa'
-------------------------------------------------------------------------
Retrieving information for 'Cerro Gordo County, Iowa'
-------------------------------------------------------------------------
Retrieving information for 'Floyd County, Iowa'
-------------------------------------------------------------------------
Retrieving information for 'Greene County, Iowa'
-------------------------------------------------------------------------
Retrieving information for 'Hardin County, Iowa'
-------------------------------------------------------------------------
Retrieving information for 'Louisa County, Iowa'
-------------------------------------------------------------------------
Retrieving information for 'Lucas County, Iowa'
-------------------------------------------------------------------------
Retrieving in

Retrieving information for 'Kingman County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Logan County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Cherokee County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Nemaha County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Pottawatomie County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Wyandotte County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Butler County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Clay County, Kansas'
-----------------------------------------------------------------

Retrieving information for 'Sedgwick County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Seward County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Republic County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Graham County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Johnson County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Douglas County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Sheridan County, Kansas'
-------------------------------------------------------------------------
Retrieving information for 'Miami County, Kansas'
-------------------------------------------------------------------

Retrieving information for 'McCreary County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Madison County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Magoffin County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Metcalfe County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Owsley County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Rowan County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Simpson County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Taylor County, Kentucky'
---------------------------------------------------

Retrieving information for 'Rockcastle County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Scott County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Warren County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Allen County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Bourbon County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Boyd County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Bullitt County, Kentucky'
-------------------------------------------------------------------------
Retrieving information for 'Crittenden County, Kentucky'
----------------------------------------------------

Retrieving information for 'Acadia Parish, Louisiana'
-------------------------------------------------------------------------
Retrieving information for 'St. Mary Parish, Louisiana'
-------------------------------------------------------------------------
Retrieving information for 'West Carroll Parish, Louisiana'
-------------------------------------------------------------------------
Retrieving information for 'St. Tammany Parish, Louisiana'
-------------------------------------------------------------------------
Retrieving information for 'Union Parish, Louisiana'
-------------------------------------------------------------------------
Retrieving information for 'Caldwell Parish, Louisiana'
-------------------------------------------------------------------------
Retrieving information for 'Caddo Parish, Louisiana'
-------------------------------------------------------------------------
Retrieving information for 'East Carroll Parish, Louisiana'
-------------------------------

Retrieving information for 'Anne Arundel County, Maryland'
-------------------------------------------------------------------------
Retrieving information for 'Baltimore County, Maryland'
-------------------------------------------------------------------------
Retrieving information for 'Frederick County, Maryland'
-------------------------------------------------------------------------
Retrieving information for 'Calvert County, Maryland'
-------------------------------------------------------------------------
Retrieving information for 'Garrett County, Maryland'
-------------------------------------------------------------------------
Retrieving information for 'Kent County, Maryland'
-------------------------------------------------------------------------
Retrieving information for 'Montgomery County, Maryland'
-------------------------------------------------------------------------
Retrieving information for 'Carroll County, Maryland'
-----------------------------------------

Retrieving information for 'Cheboygan County, Michigan'
-------------------------------------------------------------------------
Retrieving information for 'Luce County, Michigan'
-------------------------------------------------------------------------
Retrieving information for 'Kent County, Michigan'
-------------------------------------------------------------------------
Retrieving information for 'Macomb County, Michigan'
-------------------------------------------------------------------------
Retrieving information for 'Saginaw County, Michigan'
-------------------------------------------------------------------------
Retrieving information for 'Shiawassee County, Michigan'
-------------------------------------------------------------------------
Retrieving information for 'Baraga County, Michigan'
-------------------------------------------------------------------------
Retrieving information for 'Bay County, Michigan'
---------------------------------------------------------

Retrieving information for 'Pine County, Minnesota'
-------------------------------------------------------------------------
Retrieving information for 'Olmsted County, Minnesota'
-------------------------------------------------------------------------
Retrieving information for 'Cottonwood County, Minnesota'
-------------------------------------------------------------------------
Retrieving information for 'Scott County, Minnesota'
-------------------------------------------------------------------------
Retrieving information for 'Dodge County, Minnesota'
-------------------------------------------------------------------------
Retrieving information for 'Wabasha County, Minnesota'
-------------------------------------------------------------------------
Retrieving information for 'Pipestone County, Minnesota'
-------------------------------------------------------------------------
Retrieving information for 'Crow Wing County, Minnesota'
------------------------------------------

Retrieving information for 'Itawamba County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Lafayette County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Issaquena County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Jackson County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Lowndes County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Leflore County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Tunica County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Wilkinson County, Mississippi'
--------------------

Retrieving information for 'Lawrence County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Panola County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Perry County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Yalobusha County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Chickasaw County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Carroll County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'Copiah County, Mississippi'
-------------------------------------------------------------------------
Retrieving information for 'George County, Mississippi'
--------------------------

Retrieving information for 'Barry County, Missouri'
-------------------------------------------------------------------------
Retrieving information for 'Barton County, Missouri'
-------------------------------------------------------------------------
Retrieving information for 'Boone County, Missouri'
-------------------------------------------------------------------------
Retrieving information for 'Cedar County, Missouri'
-------------------------------------------------------------------------
Retrieving information for 'Cole County, Missouri'
-------------------------------------------------------------------------
Retrieving information for 'Crawford County, Missouri'
-------------------------------------------------------------------------
Retrieving information for 'Gasconade County, Missouri'
-------------------------------------------------------------------------
Retrieving information for 'Iron County, Missouri'
------------------------------------------------------------

Retrieving information for 'Custer County, Montana'
-------------------------------------------------------------------------
Retrieving information for 'Hill County, Montana'
-------------------------------------------------------------------------
Retrieving information for 'Chouteau County, Montana'
-------------------------------------------------------------------------
Retrieving information for 'Treasure County, Montana'
-------------------------------------------------------------------------
Retrieving information for 'Phillips County, Montana'
-------------------------------------------------------------------------
Retrieving information for 'Powder River County, Montana'
-------------------------------------------------------------------------
Retrieving information for 'Lincoln County, Montana'
-------------------------------------------------------------------------
Retrieving information for 'Park County, Montana'
---------------------------------------------------------

Retrieving information for 'Chase County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Butler County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Clay County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Arthur County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Custer County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Franklin County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Gage County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Hamilton County, Nebraska'
-----------------------------------------------------------

Retrieving information for 'Nuckolls County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Rock County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Banner County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Frontier County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Sherman County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Douglas County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Richardson County, Nebraska'
-------------------------------------------------------------------------
Retrieving information for 'Howard County, Nebraska'
--------------------------------------------------

Retrieving information for 'Hunterdon County, New Jersey'
-------------------------------------------------------------------------
Retrieving information for 'Burlington County, New Jersey'
-------------------------------------------------------------------------
Retrieving information for 'Morris County, New Jersey'
-------------------------------------------------------------------------
Retrieving information for 'San Juan County, New Mexico'
-------------------------------------------------------------------------
Retrieving information for 'Quay County, New Mexico'
-------------------------------------------------------------------------
Retrieving information for 'Lincoln County, New Mexico'
-------------------------------------------------------------------------
Retrieving information for 'Chaves County, New Mexico'
-------------------------------------------------------------------------
Retrieving information for 'Lea County, New Mexico'
-------------------------------------

Retrieving information for 'Montgomery County, New York'
-------------------------------------------------------------------------
Retrieving information for 'Nassau County, New York'
-------------------------------------------------------------------------
Retrieving information for 'Richmond County, New York'
-------------------------------------------------------------------------
Retrieving information for 'St. Lawrence County, New York'
-------------------------------------------------------------------------
Retrieving information for 'Schuyler County, New York'
-------------------------------------------------------------------------
Retrieving information for 'Tompkins County, New York'
-------------------------------------------------------------------------
Retrieving information for 'Warren County, New York'
-------------------------------------------------------------------------
Retrieving information for 'Chenango County, New York'
----------------------------------------

Retrieving information for 'Forsyth County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Nash County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Alexander County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Bladen County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Dare County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Carteret County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Warren County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Wilson County, North Carolina'
--------

Retrieving information for 'Davidson County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Duplin County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Durham County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Hertford County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Lee County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'New Hanover County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Pender County, North Carolina'
-------------------------------------------------------------------------
Retrieving information for 'Wayne County, North Carolina'
-----

Retrieving information for 'Mahoning County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Athens County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Adams County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Scioto County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Medina County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Meigs County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Union County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Champaign County, Ohio'
-------------------------------------------------------------------------
Retrieving inf

Retrieving information for 'Delaware County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Geauga County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Guernsey County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Hocking County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Lake County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Mercer County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Ross County, Ohio'
-------------------------------------------------------------------------
Retrieving information for 'Clark County, Ohio'
-------------------------------------------------------------------------
Retrieving infor

Retrieving information for 'Johnston County, Oklahoma'
-------------------------------------------------------------------------
Retrieving information for 'Pittsburg County, Oklahoma'
-------------------------------------------------------------------------
Retrieving information for 'Pontotoc County, Oklahoma'
-------------------------------------------------------------------------
Retrieving information for 'Roger Mills County, Oklahoma'
-------------------------------------------------------------------------
Retrieving information for 'Woodward County, Oklahoma'
-------------------------------------------------------------------------
Retrieving information for 'Cherokee County, Oklahoma'
-------------------------------------------------------------------------
Retrieving information for 'Delaware County, Oklahoma'
-------------------------------------------------------------------------
Retrieving information for 'Ellis County, Oklahoma'
-----------------------------------------

Retrieving information for 'Multnomah County, Oregon'
-------------------------------------------------------------------------
Retrieving information for 'Union County, Oregon'
-------------------------------------------------------------------------
Retrieving information for 'Newport County, Rhode Island'
-------------------------------------------------------------------------
Retrieving information for 'Kent County, Rhode Island'
-------------------------------------------------------------------------
Retrieving information for 'Providence County, Rhode Island'
-------------------------------------------------------------------------
Retrieving information for 'Bristol County, Rhode Island'
-------------------------------------------------------------------------
Retrieving information for 'Washington County, Rhode Island'
-------------------------------------------------------------------------
Retrieving information for 'Delaware County, Pennsylvania'
--------------------------

Retrieving information for 'Montour County, Pennsylvania'
-------------------------------------------------------------------------
Retrieving information for 'Warren County, Pennsylvania'
-------------------------------------------------------------------------
Retrieving information for 'Columbia County, Pennsylvania'
-------------------------------------------------------------------------
Retrieving information for 'Butler County, Pennsylvania'
-------------------------------------------------------------------------
Retrieving information for 'Elk County, Pennsylvania'
-------------------------------------------------------------------------
Retrieving information for 'Jefferson County, Pennsylvania'
-------------------------------------------------------------------------
Retrieving information for 'Lancaster County, Pennsylvania'
-------------------------------------------------------------------------
Retrieving information for 'Mercer County, Pennsylvania'
--------------------

Retrieving information for 'Minnehaha County, South Dakota'
-------------------------------------------------------------------------
Retrieving information for 'Custer County, South Dakota'
-------------------------------------------------------------------------
Retrieving information for 'Hyde County, South Dakota'
-------------------------------------------------------------------------
Retrieving information for 'Clark County, South Dakota'
-------------------------------------------------------------------------
Retrieving information for 'Faulk County, South Dakota'
-------------------------------------------------------------------------
Retrieving information for 'Hutchinson County, South Dakota'
-------------------------------------------------------------------------
Retrieving information for 'Kingsbury County, South Dakota'
-------------------------------------------------------------------------
Retrieving information for 'Walworth County, South Dakota'
------------------

Retrieving information for 'Decatur County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Davidson County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Jackson County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'McNairy County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Perry County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Roane County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Bradley County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Meigs County, Tennessee'
-----------------------------------------------

Retrieving information for 'Cheatham County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Carter County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Grainger County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Giles County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Lauderdale County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Lawrence County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Madison County, Tennessee'
-------------------------------------------------------------------------
Retrieving information for 'Marion County, Tennessee'
----------------------------------------

Retrieving information for 'Lamb County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Lipscomb County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Loving County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Matagorda County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Pecos County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Robertson County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Rockwall County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Rusk County, Texas'
-------------------------------------------------------------------------
Ret

Retrieving information for 'Foard County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Freestone County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Goliad County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Hamilton County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Hardeman County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Mitchell County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Titus County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Briscoe County, Texas'
-------------------------------------------------------------------------


Retrieving information for 'Cass County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Gaines County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Gregg County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Gray County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Madison County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Lubbock County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Jack County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Kaufman County, Texas'
-------------------------------------------------------------------------
Retrieving i

Retrieving information for 'Blanco County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Jasper County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Jefferson County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Kinney County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Oldham County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Deaf Smith County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'Galveston County, Texas'
-------------------------------------------------------------------------
Retrieving information for 'El Paso County, Texas'
-----------------------------------------------------------------------

Retrieving information for 'Culpeper County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Dinwiddie County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Greene County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Hanover County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Lunenburg County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Orange County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Emporia city, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Wythe County, Virginia'
---------------------------------------------------

Retrieving information for 'Buena Vista city, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Falls Church city, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Manassas Park city, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Petersburg city, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Poquoson city, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Sussex County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Fairfax city, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Waynesboro city, Virginia'
-------------------------------------------

Retrieving information for 'Powhatan County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Rockbridge County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Shenandoah County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Fauquier County, Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Harrison County, West Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Ohio County, West Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Ritchie County, West Virginia'
-------------------------------------------------------------------------
Retrieving information for 'Calhoun County, West Virginia'
------------------------

Retrieving information for 'Benton County, Washington'
-------------------------------------------------------------------------
Retrieving information for 'Columbia County, Washington'
-------------------------------------------------------------------------
Retrieving information for 'Clark County, Washington'
-------------------------------------------------------------------------
Retrieving information for 'Asotin County, Washington'
-------------------------------------------------------------------------
Retrieving information for 'Adams County, Washington'
-------------------------------------------------------------------------
Retrieving information for 'Garfield County, Washington'
-------------------------------------------------------------------------
Retrieving information for 'Mason County, Washington'
-------------------------------------------------------------------------
Retrieving information for 'Skagit County, Washington'
-----------------------------------------

Retrieving information for 'Lafayette County, Wisconsin'
-------------------------------------------------------------------------
Retrieving information for 'Monroe County, Wisconsin'
-------------------------------------------------------------------------
Retrieving information for 'Pierce County, Wisconsin'
-------------------------------------------------------------------------
Retrieving information for 'Wood County, Wisconsin'
-------------------------------------------------------------------------
Retrieving information for 'Ashland County, Wisconsin'
-------------------------------------------------------------------------
Retrieving information for 'Calumet County, Wisconsin'
-------------------------------------------------------------------------
Retrieving information for 'Florence County, Wisconsin'
-------------------------------------------------------------------------
Retrieving information for 'Jefferson County, Wisconsin'
------------------------------------------

Retrieving information for 'Ozaukee County, Wisconsin'
-------------------------------------------------------------------------
Retrieving information for 'Portage County, Wisconsin'
-------------------------------------------------------------------------
Retrieving information for 'Richland County, Wisconsin'
-------------------------------------------------------------------------


### Get the DataFrame with the Coordinates

In [54]:
coordinates_pd.head(50)

State                                  Name  Population  Median Age  \
0      72        Corozal Municipio, Puerto Rico     34933.0        38.1   
1      72        Maunabo Municipio, Puerto Rico     11297.0        43.5   
2      72       Peñuelas Municipio, Puerto Rico     21661.0        36.4   
3      72          Ponce Municipio, Puerto Rico    148863.0        39.5   
4      72  San Sebastián Municipio, Puerto Rico     38970.0        42.0   
5      72       Toa Alta Municipio, Puerto Rico     74169.0        37.1   
6      72       Toa Baja Municipio, Puerto Rico     81905.0        38.9   
7      72        Vieques Municipio, Puerto Rico      8931.0        41.4   
8      72        Yabucoa Municipio, Puerto Rico     35025.0        41.6   
9      72         Aguada Municipio, Puerto Rico     39470.0        41.0   
10     72         Caguas Municipio, Puerto Rico    134269.0        39.7   
11     72      Canóvanas Municipio, Puerto Rico     46779.0        39.0   
12     72          Cidra Municipio, Puerto Rico     41181.0        38.7   
13     72          Coamo Municipio, Puerto Rico     39796.0        39.4   
14     72         Dorado Municipio, Puerto Rico     37722.0        39.5   
15     72       Luquillo Municipio, Puerto Rico     18952.0        39.8   
16     72   Barranquitas Municipio, Puerto Rico     29237.0        35.5   
17     72       Orocovis Municipio, Puerto Rico     21906.0        37.9   
18     72    San Lorenzo Municipio, Puerto Rico     38689.0        40.2   
19     72         Añasco Municipio, Puerto Rico     27892.0        41.3   
20     72          Ceiba Municipio, Puerto Rico     12238.0        40.9   
21     72         Ciales Municipio, Puerto Rico     17325.0        40.5   
22     72        Comerío Municipio, Puerto Rico     19914.0        38.1   
23     72        Isabela Municipio, Puerto Rico     43314.0        40.8   
24     72         Jayuya Municipio, Puerto Rico     15297.0        36.3   
25     72          Lares Municipio, Puerto Rico     27321.0        41.1   
26     72    Las Piedras Municipio, Puerto Rico     38253.0        38.1   
27     72     Río Grande Municipio, Puerto Rico     51768.0        40.2   
28     72       San Juan Municipio, Puerto Rico    355181.0        41.8   
29     72      Vega Alta Municipio, Puerto Rico     38589.0        37.8   
30     72      Vega Baja Municipio, Puerto Rico     54754.0        40.1   
31     72        Maricao Municipio, Puerto Rico      6180.0        42.7   
32     72       Aibonito Municipio, Puerto Rico     24008.0        40.7   
33     72     Guayanilla Municipio, Puerto Rico     19578.0        40.3   
34     72         Juncos Municipio, Puerto Rico     39704.0        37.6   
35     72          Lajas Municipio, Puerto Rico     23893.0        43.4   
36     72  Sabana Grande Municipio, Puerto Rico     23610.0        40.9   
37     72       Adjuntas Municipio, Puerto Rico     18525.0        39.5   
38     72          Camuy Municipio, Puerto Rico     32936.0        40.1   
39     72        Guánica Municipio, Puerto Rico     17327.0        41.4   
40     72    Hormigueros Municipio, Puerto Rico     16474.0        45.1   
41     72         Manatí Municipio, Puerto Rico     40705.0        40.9   
42     72        Naguabo Municipio, Puerto Rico     26584.0        37.0   
43     72      Naranjito Municipio, Puerto Rico     29112.0        38.7   
44     72     San Germán Municipio, Puerto Rico     32937.0        42.7   
45     72         Utuado Municipio, Puerto Rico     30209.0        41.7   
46     72        Arecibo Municipio, Puerto Rico     89550.0        41.4   
47     72         Arroyo Municipio, Puerto Rico     18504.0        38.7   
48     72         Cataño Municipio, Puerto Rico     25595.0        39.1   
49     72        Florida Municipio, Puerto Rico     12140.0        38.2   

    Household Income  Per Capita Income  Poverty Count  Poverty Rate  \
0            14752.0             7887.0        19146.0     54.807775   
1            17636.0             8

### Export the data

In [55]:
coordinates_pd.to_csv("output/coordinates_pd.csv", encoding="utf-8", index=False)